lib/SQL/Abstract/mysql.pm

```diff
@@ -35,6 +47,36 @@ sub insert {
   return wantarray ? ($sql, @bind) : $sql;
 }
 
+sub _insert_value {
+  my ($self, $column, $v) = @_;
+
+  if (ref $v eq 'HASH') {
```

Comment:
Is there a downside to supporting the same syntax as in https://github.com/jhthorsen/mojo-mysql/blob/38c16bdc6822e2039a258b46170e3874541fa921/lib/Mojo/mysql/Database.pm#L311

Like this:

```
if (ref $v eq 'HASH' and exists $param->{json}) { ... }
```

It's a bit extra to type, but it would allow it to be extended further in the future.

Reply:
A shorter answer than the one below: We don't write
`$db->query('insert into foo values (?)', {integer => 23});` either.